src/qt/optionsmodel.cpp

```diff
@@ -282,7 +283,11 @@ QVariant OptionsModel::data(const QModelIndex & index, int role) const
         case CoinControlFeatures:
             return fCoinControlFeatures;
         case DatabaseCache:
-            return settings.value("nDatabaseCache");
+            if (!gArgs.GetArg("-dbcache", "").empty())
+                return (qint64)gArgs.GetArg("-dbcache", nDefaultDbCache);
+            if (settings.contains("nDatabaseCache"))
+                return settings.value("nDatabaseCache");
+            return QVariant::fromValue(nDefaultDbCache);
```

Comment:
Seems you can return the default directly, the QVariant will be constructed from it.

Reply:
I seem to be getting this error when I try to do that:

![deepinscreenshot_select-area_20181021212834](https://user-images.githubusercontent.com/10647624/47311351-56344900-d5ee-11e8-9d96-192a12514d2a.png)

Reply:
How about casting instead as above?

Reply:
that works, and makes the code prettier! fixed :)